src/rpc/core_rpc_server.cpp

```diff
@@ -661,7 +661,7 @@ namespace cryptonote
 
       bool incremental;
       std::vector<std::pair<crypto::hash, tx_memory_pool::tx_details>> added_pool_txs;
-      bool success = m_core.get_pool_info((time_t)req.pool_info_since, allow_sensitive, max_tx_count, added_pool_txs, res.remaining_added_pool_txids, res.removed_pool_txids, incremental);
+      bool success = m_core.get_pool_info((time_t)req.pool_info_since, allow_sensitive, max_tx_count, added_pool_txs, res.remaining_added_pool_txids, res.removed_pool_txids, incremental, LEVIN_DEFAULT_MAX_PACKET_SIZE * 0.9);
```

Comment:
If we also have any blocks sent during this RPC call (i.e. when `req.requested_info == COMMAND_RPC_GET_BLOCKS_FAST::BLOCKS_AND_POOL`), then this limit is not conservative enough. We care about the message size with all data included, after serialization.

Reply:
Also, if we limit the transactions sent but set the pool clock time to what it would be if we did NOT limit the transactions sent, then there will be "gaps" in our scanning of the pool, so we need to account for that.

Reply:
Instead of `0.9`, I  can add macro in `cryptonote_config`. What ratio do you think is ideal here?

Reply:
I will think about the gap. Off the top of your head. What is a best solution to address the gap issue?

Reply:
And please keep in mind this is exact ( logically ) solution the restricted rpc is using right now. 

So if there is any inconsistency issue, it is totally applicable to that too.

Reply:
> Instead of `0.9`, I can add macro in `cryptonote_config`. What ratio do you think is ideal here?

Well, no single ratio will be good here since we have to take into account two main variables: cumulative block data size, and cumulative pool update size. We would need to estimate the size of the total RPC response (block data AND pool updates) and then try multiplying *that* by 0.9 (or some other factor) for epee field names, array/section sizes, etc. 

> I will think about the gap. Off the top of your head. What is a best solution to address the gap issue?

The best, in terms of consistency, would return the pool clock time value which of the last returned pool event, and make sure we only return pool updates in contiguous pool time chunks. 

> And please keep in mind this is exact ( logically ) solution the restricted rpc is using right now.
> 
> So if there is any inconsistency issue, it is totally applicable to that too.

Not really, the restricted RPC limits are very much soft limits put much lower than other hard, technical limits. Also, the restricted RPC limits tend to be deterministic before we fetch the data (e.g. number of entries) as opposed to the number of bytes of a serialized response, which isn't known to the RPC endpoint methods.

Reply:
> The best, in terms of consistency, would return the pool clock time value which of the last returned pool event, and make sure we only return pool updates in contiguous pool time chunks.

This applicable to restricted and unrestricted.

Reply:
> Not really, the restricted RPC limits are very much soft limits put much lower than other hard, technical limits. Also, the restricted RPC limits tend to be deterministic before we fetch the data (e.g. number of entries) as opposed to the number of bytes of a serialized response, which isn't known to the RPC endpoint methods.

Can you elaborate this with an example? Since it is not obvious to me how that makes it different logically.

Reply:
> This applicable to restricted and unrestricted.

It wouldn't be applicable to unrestricted because an unrestricted fetch will return a daemon time for which pool updates are consistent. Whereas the way the code is structured right now, if we don't include ALL pool updates since `pool_info_since` BUT return the current wall time in `daemon_time`, then the wallet will necessarily miss pool updates.

Reply:
@jeffro256 based on discussion in research meeting with jberman, do you think we can go ahead and submit this PR to main repo? 

P.S. Logs should be available here soon: https://github.com/monero-project/meta/issues/1078

Reply:
There's two major things that need updating with this PR: 1) the limit for pool calculation needs to include block data since it's all returned in the same response and 2) the `remaining_added_txids` vector calculation needs better failure handling to actually be consistent.